File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.cpp
class Solution {
public:
    long long countCommas(long long n) {
        long long cnt=0;
        long long start=1000;
        long long commas=1;
        while(start<=n){
            long long end=start*1000-1;
            if(end>n){
                end=n;
            }
            cnt+=(end-start+1)*commas;
            start=start*1000;
            commas++;
        }
        return cnt;
    }
};

File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.py
class Solution:
    def countCommas(self, n: int) -> int:
        cnt,commas,strt=0,1,1000
        while strt<=n:
            end=strt*1000-1
            if end>n:
                end=n
            cnt+=(end-strt+1)*commas
            strt=strt*1000
            commas+=1
        return cnt